scripts/mbedtls_dev/psa_information.py

```diff
@@ -104,3 +90,94 @@ def read_psa_interface(self) -> macro_collector.PSAMacroEnumerator:
         self.remove_unwanted_macros(constructors)
         constructors.gather_arguments()
         return constructors
+
+
+class TestCase(test_case.TestCase):
+    """A PSA test case with automatically inferred dependencies.
+
+    For mechanisms like ECC curves where the support status includes
+    the key bit-size, this class assumes that only one bit-size is
+    involved in a given test case.
+    """
+
+    # Use a class variable to cache the set of implemented dependencies.
+    # Call read_implemented_dependencies() to fill the cache.
+    _implemented_dependencies = None #type: Optional[FrozenSet[str]]
+
+    DEPENDENCY_SYMBOL_RE = re.compile(r'\bPSA_WANT_\w+\b')
+    @classmethod
+    def _yield_implemented_dependencies(cls) -> Iterator[str]:
+        for filename in ['include/psa/crypto_config.h',
+                         'include/mbedtls/config_psa.h']:
```

Comment:
This isn't quite right. For example, mechanisms that are commented out in `crypto_config.h` will be considered as implemented, and conversely if you run this script with an edited `crypto_config.h` then some supported mechanisms might not be considered to be implemented. I'm considering some solutions, but unsure yet whether to apply them here or in a follow-up, because this pull request is already pretty long and I think I've reached a decent stepping point.

Reply:
The overall issue remains https://github.com/Mbed-TLS/mbedtls/issues/5390.